Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -718,3 +718,45 @@ void JitArm64::addcx(UGeckoInstruction inst)
 			ComputeRC(gpr.R(d), 0);
 	}
 }
+
+void JitArm64::slwx(UGeckoInstruction inst)
+{
+	INSTRUCTION_START
+	JITDISABLE(bJITIntegerOff);
+
+	int a = inst.RA, b = inst.RB, s = inst.RS;
+
+	if (gpr.IsImm(b) && gpr.IsImm(s))
+	{
+		u32 i = gpr.GetImm(s), j = gpr.GetImm(b);
+		gpr.SetImmediate(a, (j & 0x20) ? 0 : i << (j & 0x1F));
+
+		if (inst.Rc)
+			ComputeRC(gpr.GetImm(a), 0);
+	}
+	else if (gpr.IsImm(b))
+	{
+		u32 i = gpr.GetImm(b);
+		if (i & 0x20)
+		{
+			gpr.SetImmediate(a, 0);
+			if (inst.Rc)
+				ComputeRC(0, 0);
+		}
+		else
+		{
+			LSL(gpr.R(a), gpr.R(s), i & 0x1F);
+			if (inst.Rc)
+				ComputeRC(gpr.R(a), 0);
+		}
+	}
+	else
+	{
+		gpr.BindToRegister(a, a == b || a == s);
+
+		LSLV(EncodeRegTo64(gpr.R(a)), EncodeRegTo64(gpr.R(s)), EncodeRegTo64(gpr.R(b)));
```

Comment:
It'll actually result as a 64bit shift, so we're in the clear since the bottom 32bits will be shifted out to the top 32bits, resulting in zero.